File: Algorithms/Searching/Python/JumpSearch.py
import math


class JumpSearch:
    arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 14]
    n = len(arr)
    find = 5

    def search(self, arr, find):
        m = int(math.floor(math.sqrt(len(arr))))

        currentLastIndex = m

        while currentLastIndex < len(arr) and find >= arr[currentLastIndex]:
            currentLastIndex += m

        i = currentLastIndex - m

        while i < currentLastIndex and i < len(arr):
            if arr[i] == find:
                return i
            i += 1

        return -1


obj = JumpSearch()

pos = obj.search(obj.arr, find=obj.find)

if pos != -1:
    print("Number " + str(obj.find) + " at position " + str(pos + 1))
else:
    print("Number " + str(obj.find) + " not found")

File: Algorithms/Searching/Java/JumpSearch.java
package Algorithms.Searching.Java;

public class JumpSearch {

    static int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14};
    static int find = 14;

    public static void main(String[] args) {

        int position = search(arr);

        if (position != -1) {
            System.out.println("Element found at position " + (position + 1));
        } else {
            System.out.println("Element not found");
        }

    }

    public static int search(int[] arr) {

        int m = (int) Math.floor(Math.sqrt(arr.length));

        int lastIndex = m;

        while (lastIndex < arr.length && find >= arr[lastIndex]) {
            lastIndex += m;
        }

        for (int i = lastIndex - m; i < arr.length && i < lastIndex; i++) {
            if (arr[i] == find) {
                return i;
            }
        }


        return -1;
    }

}
